docs/topics/native/native-dynamic-libraries.md

```diff
@@ -302,117 +338,77 @@ typedef struct {
+The code uses anonymous structure declarations. Here, `struct { ... } foo` declares a field in the outer struct of that
+anonymous structure type, the type with no name.
 
-### Your library functions
+C doesn't support objects either, so function pointers are used to mimic object semantics. A function pointer is declared
+as `RETURN_TYPE (* FIELD_NAME)(PARAMETERS)`.
 
-Let's take a look at the `kotlin.root.example` field, it
-mimics the package structure of our Kotlin code with a `kotlin.root.` prefix.
+There is a `libnative_kref_example_Clazz` field that represents the `Clazz` from Kotlin. The `libnative_KULong` is
+accessible with the `memberFunction` field. The only difference is that the `memberFunction` accepts a `thiz` reference as
+the first parameter. The C language doesn't support objects, which is why `thiz` pointer should be passed explicitly.
 
-There is a `kotlin.root.example.Clazz` field that 
-represents the `Clazz` from Kotlin. The `Clazz#memberFunction` is
-accessible with the `memberFunction` field. The only difference is that 
-the `memberFunction` accepts a `this` reference as the first parameter. 
-The C language does not support objects, and this is the reason to pass a
-`this` pointer explicitly.
+There is a constructor in the `Clazz` field (aka `libnative_kref_example_Clazz_Clazz`), which is the constructor function
+to create an instance of the `Clazz`.
 
-There is a constructor in the `Clazz` field (aka `kotlin.root.example.Clazz.Clazz`),
-which is the constructor function to create an instance of the `Clazz`.
+Kotlin `object Object` is accessible as `libnative_kref_example_Object`. There's the `_instance` function to get the only
+instance of the object.
 
-Kotlin `object Object` is accessible as `kotlin.root.example.Object`. There is 
-the `_instance` function to get the only instance of the object.
+Properties are translated into functions. The `get_` and `set_` prefixes name the getter and the setter functions,
+respectively. For example, the read-only property `globalString` from Kotlin is turned into a `get_globalString`
+function in C.
 
-Properties are translated into functions. The `get_` and `set_` prefix
-is used to name the getter and the setter functions respectively. For example, 
-the read-only property `globalString` from Kotlin is turned 
-into a `get_globalString` function in C. 
+Global functions `forFloats`, `forIntegers`, and `strings` are turned into functions pointers in the `libnative_kref_example`
+anonymous struct.
 
-Global functions `forInts`, `forFloats`, or `strings` are turned into the functions pointers in
-the `kotlin.root.example` anonymous struct.
- 
 ### Entry point
 
-You can see how the API is created. To start with, you need to initialize the 
-`libnative_ExportedSymbols` structure. Let's take a look at the latest part 
-of the `libnative_api.h` for this:
+You can see how the API is created. To start with, you need to initialize the `libnative_ExportedSymbols` structure.
```

Comment:
Hmm 🤔 
This section is a bit difficult to follow in my opinion. Lots of things are happening at the same time - The user can see how the API created - then they have to initialize something, and then they need to look at the final part of the `libnative.api.h`.
 
Could it be cleaned up by saying something like: 
You can initialize the API using the `libnative_ExportedSymbols` structure as seen in the the final part of the `libnative_api.h`:

Or:

You can see how the API is initialized using the `libnative_ExportedSymbols` structure in the final part of the `libnative_api.h`:

Reply:
These first two sentences just connect all the previous header file description with this last part. But I can see how it maybe confusing, so I created a new version

Reply:
Nice - it reads much better now thank you! 👍